src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -226,36 +233,30 @@ private void AnalyzeDirectory(DirectoryInfo directoryPath, FileInfo configuratio
                 Console.WriteLine(Environment.NewLine + Environment.NewLine + $"Directory: {directoryPath}");
 
                 int numOfSuccesses = 0;
-                using (IReportWriter reportWriter = this.GetReportWriter(reportFormat.ToString(), outputFilePath, directoryPath.FullName))
+               
+                var filesFailed = new List<FileInfo>();
+                foreach (FileInfo file in filesToAnalyze)
                 {
-                    var filesFailed = new List<FileInfo>();
-                    foreach (FileInfo file in filesToAnalyze)
+                    int res = AnalyzeTemplate(file, null, configurationsFilePath, reportFormat, outputFilePath, runTtk, verbose, false, reportWriter, false, logger);
+                    if (res == 0)
                     {
-                        int res = AnalyzeTemplate(file, null, configurationsFilePath, reportFormat, outputFilePath, runTtk, verbose, false, reportWriter, false, logger);
-                        if (res == 0)
-                        {
-                            numOfSuccesses++;
-                        }
-                        else if (res == 1)
-                        {
-                            filesFailed.Add(file);
-                        }
+                        numOfSuccesses++;
                     }
-
-                    Console.WriteLine(Environment.NewLine + $"Analyzed {numOfSuccesses} file(s).");
-                    if (filesFailed.Count > 0)
+                    else if (res == 1)
                     {
-                        logger.LogError("Unable to analyze {numFilesFailed} file(s):", filesFailed.Count);
-                        foreach (FileInfo failedFile in filesFailed)
-                        {
-                            logger.LogError("\t{failedFile}", failedFile);
-                        }
+                        filesFailed.Add(file);
                     }
                 }
+
+                Console.WriteLine(Environment.NewLine + $"Analyzed {numOfSuccesses} file(s).");
+                if (filesFailed.Count > 0)
+                {
+                    logger.LogError($"Unable to analyze {filesFailed.Count} file(s): {string.Join(", ", filesFailed)}");
+                }
             }
-            catch (Exception exp)
+            catch (Exception exception)
             {
-                logger.LogError(GetExceptionMessage(exp));
+                logger.LogError(exception, "An exception occurred while analyzing the directory provided");
```

Comment:
Why did you remove reporting of the inner exception?

Reply:
What about in the console output scenario? Or in future output type scenarios? Will each output type have to account for iterating through the inner exceptions?

Reply:
For the console output it's handled by the general logger library, and for the SARIF output is handled by the SARIF logger library

Reply:
@yane3628 Is it okay if I add an issue to improve the formatting of the console output? I think it's out of scope for this PR

Reply:
Yea, I think making an issue to solve later is OK.